cfripper/model/principal_checking_rule.py

```diff
@@ -22,12 +22,21 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._valid_principals = None
 
+    def _get_whitelist_from_config(self, services: List[str] = None) -> Set[str]:
+        if services is None:
+            services = self._config.aws_service_accounts.keys()
+
+        unique_list = set()
+        for service in services:
+            unique_list |= set(self._config.aws_service_accounts[service])
```

Comment:
Do we need to add more safety here and check that `service` is in `self._config.aws_service_accounts` first, or is it the  case that this is never going to realistically happen?

Reply:
It shouldn't happen this case :)